packages/flutter_tools/lib/src/web/file_generators/js/flutter.js

```diff
@@ -156,45 +156,40 @@ _flutter.loader = null;
     }
 
     /**
-     * Returns the latest service worker for the given `serviceWorkerRegistrationPromise`.
+     * Returns the latest service worker for the given `serviceWorkerRegistration`.
      *
      * This might return the current service worker, if there's no new service worker
      * awaiting to be installed/updated.
      *
-     * @param {Promise<ServiceWorkerRegistration>} serviceWorkerRegistrationPromise
+     * @param {ServiceWorkerRegistration} serviceWorkerRegistration
      * @returns {Promise<ServiceWorker>}
      */
-    async _getNewServiceWorker(serviceWorkerRegistrationPromise) {
-      const reg = await serviceWorkerRegistrationPromise;
-
-      if (!reg.active && (reg.installing || reg.waiting)) {
+    async _getNewServiceWorker(serviceWorkerRegistration) {
+      if (!serviceWorkerRegistration.active && (serviceWorkerRegistration.installing || serviceWorkerRegistration.waiting)) {
         // No active web worker and we have installed or are installing
         // one for the first time. Simply wait for it to activate.
         console.debug("Installing/Activating first service worker.");
-        return reg.installing || reg.waiting;
-      } else if (!reg.active.scriptURL.endsWith(serviceWorkerVersion)) {
+        return serviceWorkerRegistration.installing || serviceWorkerRegistration.waiting;
+      } else if (!serviceWorkerRegistration.active.scriptURL.endsWith(serviceWorkerVersion)) {
```

Comment:
I think this regresses your own fix of `serviceWorkerVersion` not being a global?

Reply:
no, I just renamed variables here, `serviceWorkerVersion` is still the same, from the global context